### PR TITLE
remarks glossary integration TM-2697

### PR DESCRIPTION
**Pull request by SidelineCory24**
To-Do:

- [x] Integrate FE with Remarks Glossary EP
- [x] Displays for AO and CDO

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
image
remarks aren't supposed to have default text in the input boxes - i think what you have marked as input text is just the regular text

**Comment by SidelineCory24** [comment]
> image remarks aren't supposed to have default text in the input boxes - i think what you have marked as input text is just the regular text

Discussed offline, disabling input for now.

File: src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx [hunk 1]
```diff
@@ -46,32 +48,33 @@ const RemarksGlossary = ({ onRemarkClick, remarks }) => {
...
         return (
-          <div key={category.rmrkrccode}>
-            <div className={`remark-category remark-category--${category.rmrkrccode}`}>{category.rcdesctext}</div>
+          <div key={category.code}>
+            <div className={`remark-category remark-category--${category.code}`}>{category.desc_text}</div>
             <ul>
               {remarksInCategory.map(r => {
-                const hasTextInput = has(r, 'textInputValue');
+                // still need indicator to come through for input
+                // const hasTextInput = has(r, 'text');
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx:59]: we can default to false instead and say why, like you have in the comment, that way our code is still there an uncommented, just defaulting to not show

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 2]
```diff
@@ -55,9 +55,13 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '' }, ref) => {
   // need to update once fully integrated
   const { data, error, loading /* , retry */ } = useDataLoader(api().get, `/fsbid/assignment_history/${perdet}/`);
   const client_data = useDataLoader(api().get, `/fsbid/client/${perdet}/`);
+  const remarks = useDataLoader(api().get, '/fsbid/agenda/remarks/');
+  const remarkCategories = useDataLoader(api().get, '/fsbid/agenda/remark-categories/');
 
   const assignments = get(data, 'data') || [];
   const languages = get(client_data, 'data.data.languages') || [];
+  const remarks_data = get(remarks, 'data.data.results') || [];
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:63]: ❤️

File: src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx [hunk 3]
```diff
@@ -46,32 +48,33 @@ const RemarksGlossary = ({ onRemarkClick, remarks }) => {
...
     <div className="usa-grid-full remarks-glossary-container">
-      {remarksCategories.map(category => {
-        const remarksInCategory = orderBy(remarks$.filter(f => f.rmrkrccode === category.rmrkrccode), 'rmrkordernum');
+      {remarkCategories$$.map(category => {
+        const remarksInCategory = orderBy(remarks$.filter(f => f.rc_code === category.code), 'order_num');
         return (
-          <div key={category.rmrkrccode}>
-            <div className={`remark-category remark-category--${category.rmrkrccode}`}>{category.rcdesctext}</div>
+          <div key={category.code}>
+            <div className={`remark-category remark-category--${category.code}`}>{category.desc_text}</div>
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx:55]: 🦜

File: src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx [hunk 4]
```diff
@@ -46,32 +48,33 @@ const RemarksGlossary = ({ onRemarkClick, remarks }) => {
 
   return (
     <div className="usa-grid-full remarks-glossary-container">
-      {remarksCategories.map(category => {
-        const remarksInCategory = orderBy(remarks$.filter(f => f.rmrkrccode === category.rmrkrccode), 'rmrkordernum');
+      {remarkCategories$$.map(category => {
+        const remarksInCategory = orderBy(remarks$.filter(f => f.rc_code === category.code), 'order_num');
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx:52]: Now that we are integrating with the EPs, let's add a little bit of error handling, just a Pydash get where we are grabbing data is enough for now

Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemResearchPane/RemarksGlossary/RemarksGlossary.jsx:52]: discussed offline - defining in the proptypes

**Review by elizabeth-jimenez — APPROVED** [review]
Remark-able 😏

**Pull request merged**